webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,231 @@
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ showResourceType, setShowResourceType ] = useState(true);
+  const [ resourceType, setResourceType ] = useState<ResourceType | undefined>
+  (storage.getWithDefault(STORAGE_KEY, { type: undefined }).type);
+  const [ form ] = Form.useForm();
+
+  const fetchTemplates = useCallback(async () => {
+    try {
+      setTemplates(await getTaskTemplates({}));
+    } catch {}
+  },[]);
+
+  useEffect(() => {
+    fetchTemplates();
+  }, [ fetchTemplates ]);
+
+  const fetchResourcePools = useCallback(async () => {
+    try {
+      setResourcePools( await getResourcePools({}));
+    } catch {}
+  }, []);
+
+  useEffect(() => {
+    fetchResourcePools();
+  }, [ fetchResourcePools ]);
+
+  const fetchConfig = useCallback(async () => {
+    try {
+      const values: NotebookConfig = form.getFieldsValue(true);
+      const config = await previewNotebook(
+        values.type === ResourceType.CPU ? 0 : values.slots,
+        values.template === 'default'? undefined : values.template,
+        values.name,
+        values.pool,
+      );
+      form.setFieldsValue({ config: yaml.dump(config) });
+    } catch {}
+  }, [ form ]);
+
+  useEffect(()=> {
+    if (showFullConfig){
+      fetchConfig();
+    }
+  }, [ showFullConfig, fetchConfig ]);
+
+  const storeConfig = useCallback((_, values: NotebookConfig) => {
+    delete values.name;
+    storage.set(STORAGE_KEY,values);
+  }, [ storage ]);
+
+  const handleSecondary = useCallback(() => {
+    if (showFullConfig) {
+      setShowFullConfig(show => !show);
+    } else {
+      form.validateFields().then(() => setShowFullConfig(show => !show)).catch();
+    }
+  },[ form, showFullConfig ]);
+
+  const handleCreateEnvironment = useCallback(
+    (values) =>{
+      if (showFullConfig) {
+        launchNotebook(yaml.load(form.getFieldValue('config')) as RawJson);
+      } else {
+        launchNotebook(
+          undefined,
+          resourceType === ResourceType.CPU ? 0 : values.slots,
+          values.template === 'default' ? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      if (onLaunch) onLaunch();
+    },
+    [ showFullConfig, form, onLaunch, resourceType ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((e: string) => {
+    const pool = resourcePools.find(pool => pool.name === e);
+    if (pool){
+      if (pool.cpuContainerCapacityPerAgent > 0 && pool.slotsPerAgent && pool.slotsPerAgent > 0) {
+        setShowResourceType(true);
+      } else if (pool.cpuContainerCapacityPerAgent > 0) {
+        setResourceType(ResourceType.CPU);
+        setShowResourceType(false);
+      } else if (pool.slotsPerAgent && pool.slotsPerAgent > 0) {
+        setResourceType(ResourceType.GPU);
+        setShowResourceType(false);
+      }
```

Comment:
**suggestion:** define the checks as a variable so do you don't perform the same check multiple times. So instead of:
```
if (a > 0 && b > 0) {
} else if (a > 0) {
} else if (b > 0) {
}
```
do:
```
const aNotZero = a > 0;
const bNotZero = b > 0;
if (aNotZero && bNotZero) {
} else if (aNotZero) {
} else if (bNotZero) {
}
```